### PR TITLE
`apiUrl` should be able to apply `path` #10699

### PR DESCRIPTION
**Pull request by anatol-sialitski**

### CODE REVIEW

File: modules/lib/lib-portal/src/main/resources/lib/xp/portal.ts [hunk 1]
```diff
@@ -617,15 +626,33 @@ interface ApiUrlHandler {
...
+        application,
+        api,
+        type = 'server',
+        path = [],
+        params,
+    } = urlParams ?? {};
+
     const bean = __.newBean<ApiUrlHandler>('com.enonic.xp.lib.portal.url.ApiUrlHandler');
-    return bean.createUrl(__.toScriptValue(params));
+    bean.setPath(([] as string[]).concat(path));
```
Comment (rymsha) [inline modules/lib/lib-portal/src/main/resources/lib/xp/portal.ts:650]: Single string must not be interpreted as single element array.
We should reserve string to preencoded path.

File: modules/lib/lib-portal/src/main/resources/lib/xp/examples/portal/apiUrl.js [hunk 2]
```diff
@@ -8,14 +8,22 @@ const url = portalLib.apiUrl({
     params: {
         a: 1,
         b: 2
-    }
+    },
+    path: 'mypath'
```
Comment (rymsha) [inline modules/lib/lib-portal/src/main/resources/lib/xp/examples/portal/apiUrl.js:12]: `mypath/subpath` would be a better example

**Review by rymsha — CHANGES_REQUESTED** [review]

File: modules/lib/lib-portal/src/main/resources/lib/xp/examples/portal/apiUrl.js (hunk 2 above)
Comment (anatol-sialitski) [reply modules/lib/lib-portal/src/main/resources/lib/xp/examples/portal/apiUrl.js:12]: I don't agree with that. You left the comment https://github.com/enonic/xp/pull/10698#discussion_r1771623612 where your idea is better, than you wrote here.

Comment (rymsha) [inline modules/lib/lib-portal/src/main/resources/lib/xp/examples/portal/apiUrl.js:12]:
```suggestion
    path: 'mypath/subpath'
```

File: modules/lib/lib-portal/src/main/resources/lib/xp/portal.ts [hunk 3]
```diff
@@ -642,12 +643,18 @@ export function apiUrl(urlParams: ApiUrlParams): string {
...
-        path = [],
+        path,
         params,
     } = urlParams ?? {};
 
     const bean = __.newBean<ApiUrlHandler>('com.enonic.xp.lib.portal.url.ApiUrlHandler');
-    bean.setPath(([] as string[]).concat(path));
+    if (path) {
+        if (Array.isArray(path)) {
+            bean.setPath(__.toScriptValue(([] as string[]).concat(path)));
```
Comment (rymsha) [inline modules/lib/lib-portal/src/main/resources/lib/xp/portal.ts:653]:
```suggestion
            bean.setPath(__.toScriptValue(path));
```

File: modules/portal/portal-api/src/main/java/com/enonic/xp/portal/url/ApiUrlParams.java [hunk 4]
```diff
@@ -46,9 +46,16 @@ public ApiUrlParams api( final String value )
         return this;
     }
 
-    public ApiUrlParams path( final List<String> value )
+    public ApiUrlParams path( final Object value )
     {
-        this.path = value;
+        if ( value instanceof ScriptValue && ( (ScriptValue) value ).isArray() )
```
Comment (rymsha) [inline modules/portal/portal-api/src/main/java/com/enonic/xp/portal/url/ApiUrlParams.java:51]: it is work of ApiUrlHander.
Also one pathSegments should accept List, another path - accepts path

**Review by rymsha — CHANGES_REQUESTED** [review]

**Review by rymsha — APPROVED** [review]

**Pull request merged**